openpype/hosts/blender/api/workio.py

```diff
@@ -87,3 +94,35 @@ def work_root(session: dict) -> str:
     if scene_dir:
         return str(Path(work_dir, scene_dir))
     return work_dir
+
+
+def check_workfile_up_to_date() -> bool:
+    """Check if the current workfile is out of date.
+
+    This is based on last modification date, so if a user modifies an out of
+    date workfile, this will return `False`. Also, in case of partial publish,
+    this will return `True`.
+
+    Returns:
+        bool: True if the current workfile is up to date.
+    """
+
+    # Get date and time of the latest published workfile
+    last_published_version = get_last_version_by_subset_name(
+        get_current_project_name(),
+        f"workfile{get_current_task_name()}",
```

Comment:
The subset naming can be customized as far as I know through settings - as such, it wouldn't be entirely safe to assume the subset name matches exactly this.